### PR TITLE
Don't believe any of us will be proud of this code tomorrow.

### PR DESCRIPTION
**Pull request by rynkwn**
The sell endpoint.

### CODE REVIEW

**Comment by kevinmichaelchen** [comment]
Seems like you're models are getting a bit out of control. Not sure this is Rails' strongest attribute, or if I quite understand the code anymore, but have you considered [STI](http://www.informit.com/articles/article.aspx?p=2220311&seqNum=4)?

Some useful links:
https://about.futurelearn.com/blog/refactoring-rails-sti/
http://blog.arkency.com/2013/07/sti/

**Comment by rynkwn** [comment]
So the issue with STI in this case is that the properties among the different models are all, potentially, very different. The way it's set up right now is that there's a polymorphic relationship between the Item, and a Subitem (where the Item is general properties common to all items, and the Subitem is the table containing the specific attributes of a particular item. For example, A shirt would have things like price and quantity common to all items, but unlike makeup, a shirt has a size and material.)

I will also say that I'm not proud of the current organization though. For all the models that were created, organization and future ease of use were not emphasized in this pull request though. A tech team member was put under a deadline last minute, so I helped out a bit to making sure that deadline was met without taking too much time. Because of this, and because I was also leaving for home the next day, the focus was on allowing for a specific function, rather than intelligently creating the structure.

Also Kevin, I want you to know that both Shannon and I have left the startup for various reasons. (Though Batjin is carrying on with it). I think I speak for both Shannon and myself though when I say that we were incredibly glad to have had you over the summer (and before). And that the fact that you've kept up with us despite having your own job is something we're beyond grateful for. You may not have met all of our team members (both tech team and beyond) this summer, but I can tell you that they all know about you, and that your commit comments are something they all notice and appreciate.

tl;dr You're an awesome person. (And in case she may have anything to add, @shannonshsu )
